crates/red_knot_python_semantic/src/types/context.rs

```diff
@@ -82,61 +82,68 @@ impl<'db> InferContext<'db> {
     ) where
         T: Ranged,
     {
-        let Some(builder) = self.report_lint(lint) else {
+        let Some(builder) = self.report_lint(lint, ranged) else {
             return;
         };
-        let mut reporter = builder.build("");
-        let diag = reporter.diagnostic();
-        let span = Span::from(self.file).with_range(ranged.range());
-        diag.annotate(Annotation::primary(span).message(message));
+        let mut diag = builder.build("");
+        diag.set_primary_message(message);
     }
 
-    /// Optionally return a reporter builder for adding a lint diagnostic.
+    /// Optionally return a builder for a lint diagnostic guard.
     ///
-    /// If the current context believes a diagnostic should be reported for the
-    /// given lint, then a reporter builder is returned that enables building
-    /// a diagnostic. The severity of the diagnostic returned is automatically
-    /// determined by the given lint and configuration. The message given is
-    /// used to construct the initial diagnostic and should be considered the
-    /// "primary message" of the diagnostic. (i.e., If nothing else about the
+    /// If the current context believes a diagnostic should be reported for
+    /// the given lint, then a builder is returned that enables building a
+    /// lint diagnostic guard. The guard can then be used, via its `DerefMut`
+    /// implementation, to directly mutate a `Diagnostic`.
+    ///
+    /// The severity of the diagnostic returned is automatically determined
+    /// by the given lint and configuration. The message given is used
+    /// to construct the initial diagnostic and should be considered the
+    /// "top-level message" of the diagnostic. (i.e., If nothing else about the
```

Comment:
I think the message part here is outdated. Or to what message does the comment here refer to?

Reply:
Yeah I flubbed this. Thanks. Fixed. It's referring to the message passed to `build()`.